kernel/arch/riscv/process.c

```diff
@@ -96,12 +445,110 @@ int setup_process(struct elf_program_info *pinfo,
                   struct task **ti_ref,
                   regs_t *r)
 {
-   NOT_IMPLEMENTED();
```

Comment:
I moved this function alone with others to the portable `process.c` code, to help you with this port.